application/libraries/Gateways/Wirecard.php

```diff
@@ -0,0 +1,159 @@
+<?php defined('BASEPATH') or exit('No direct script access allowed');
+
+use Moip\Moip;
+use Moip\Auth\BasicAuth;
+use Moip\Helper\Filters;
+use Moip\Resource\OrdersList;
+use Moip\Helper\Pagination;
+
+class Wirecard
+{
+
+    public function getPayment(
+        $access_token,
+        $public_key,
+        $nameClient,
+        $emailClient,
+        $cpfClient,
+        $phoneClient,
+        $addressRua,
+        $addressNumero,
+        $addressBairro,
+        $addressCidade,
+        $addressEstado,
+        $addressCep,
+        $idOs,
+        $title,
+        $unit_price,
+        $quantity
+    ) {
+
+        $token = $access_token;
+        $key = $public_key;
+
+        $numero = $phoneClient;
+        $novoNumero = preg_replace("/[^0-9]/", "", $numero);
+        $ddd_cliente = preg_replace('/\A.{2}?\K[\d]+/', '', $novoNumero);
+        $numero_cliente = preg_replace('/^\d{2}/', '', $novoNumero);
+
+        $addressCep = preg_replace("/[^0-9]/", "", $addressCep);
+
+        $cpfClient = preg_replace("/[^0-9]/", "", $cpfClient);
+
+        $new_unit_price = intval(preg_replace('/[^0-9]/', '', number_format($unit_price, 2, '.', '')));
+
+        $moip = new Moip(new BasicAuth($token, $key), Moip::ENDPOINT_SANDBOX);
```

Comment:
Aqui vai precisar ajudar pra quando for produção da mesma forma do wirecard.

Reply:
Quando o desenvolvimento for finalizado, não precisa deixar em SandBox. Pode fica já em definitivo o ENDPOINT_PRODUCTION.